src/lib/grammar_ast.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub struct GrammarAST {
```

Comment:
Okay, that makes sense I guess. I was just confused about the name, as AST always means abstract syntax tree to me. What does it stands for in this case? Maybe we can find a less ambiguous name?